### PR TITLE
Scroll to recently added metadata row

### LINKED ISSUE #5929: Show and focus always new metadata fields in the metadataeditor
**Is your feature request related to a problem? Please describe.**
If a new metadata field is added in the metadata editor, it is necessary to scroll down to its position, which is determined in the ruleset. In processes with much metadata, especially metadata groups, this is cumbersome. 

**Describe the solution you'd like**
If a new metadata field is added in the metadata editor, its position should be shown directly. Regarding metadata groups, the first sub-metadata field should be shown. Thus, it is possible to enter the values quickly.

The metadata field should be "on top" in the metadata frame. It should be possible to see as much sub-metadata fields of a metadata group as possible without additional scrolling. See screenshot in section _Addition context_.

**Additional context**
This can be seen as alternative for:
- #5886 

~~Here, it is necessary, to click the button "Übernehmen", which is not necessary it the new metadata field is directly shown.~~
Edit: This is not correct - there is no difference in the number of clicks. See comments below. 

Position of a metadata group
positionNewMetadata

### PR DESCRIPTION
**Pull request by thomaslow**
Fixes #5929 

Adds Javascript to scroll to the recently added metadata row and highlights it by changing its background color.

Todo

- [x] Clean up code
- [x] Apply the same logic to other metadata tables (e.g. when creating a new process)
- [x] Selenium Test

Blocked by

- #6291

Demo

https://github.com/user-attachments/assets/f07bbfd1-fa0c-4d75-9085-cb8bee49f00a

### CODE REVIEW

**Comment by solth** [comment]
@thomaslow please rebase against current master. There seems to be a test failing - `MetadataST.linkPageToNextDivision` - after #6255 was merged.

File: Kitodo/src/main/webapp/WEB-INF/resources/js/metadata_table.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        if (row.length === 1) {
+            row.addClass("focusedRow");
+            row.get(0).scrollIntoView();
+            row.find("input:enabled:visible,textarea:enabled:visible").first().focus();
+        }
+    }
+
+}
+
+// register class with global metadataTabel namespace
```
[the remark this answers is not in the document]
Comment (thomaslow) [reply Kitodo/src/main/webapp/WEB-INF/resources/js/metadata_table.js]: Done

File: Kitodo/src/main/webapp/WEB-INF/resources/js/metadata_table.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+var metadataTable = metadataTable || {};
+metadataTable.focusMetadataRow = new FocusMetadataRow();
```
[the remark this answers is not in the document]
Comment (thomaslow) [reply Kitodo/src/main/webapp/WEB-INF/resources/js/metadata_table.js]: Done

File: Kitodo/src/test/java/org/kitodo/selenium/MetadataST.java [hunk 3]
```diff
@@ -208,6 +208,11 @@ public void showPaginationByDefaultTest() throws Exception {
         Pages.getUserEditPage().setPaginationToShowByDefault();
         Pages.getProcessesPage().goTo().editMetadata(MockDatabase.MEDIA_RENAMING_TEST_PROCESS_TITLE);
         assertTrue(Pages.getMetadataEditorPage().isPaginationPanelVisible());
+        // disable pagination again to prevent conflicts with other tests (when interacting with metadata table)
+        Pages.getMetadataEditorPage().closeEditor();
+        Pages.getUserEditPage().setPaginationToShowByDefault();
```
[the remark this answers is not in the document]
Comment (thomaslow) [reply Kitodo/src/test/java/org/kitodo/selenium/MetadataST.java]: Renamed it to `togglePaginationToShowByDefault`.

**Comment by thomaslow** [comment]
> Perhaps we could adjust the color of the focused row in the future to something that is already used in the Kitodo color scheme.

Of course, my color choice is debatable. There is no usable existing color in the color palette of `pattern-library.css`. Maybe a slightly darker blue would be better?

image

The regular background uses the CSS variable `--trans-blue`. I added a variable `--trans-blue-highlight`.

**Comment by solth** [comment]
> Maybe a slightly darker blue would be better?

Yes, I think this works a lot better. Thanks for the update.

**Pull request merged**